main.c

```diff
@@ -235,20 +236,84 @@ static bool path_exists(const char *path) {
 	return path && access(path, R_OK) != -1;
 }
 
-static const char *get_output_dir(void) {
-	static const char *output_dirs[] = {
-		"GRIM_DEFAULT_DIR",
-		"XDG_PICTURES_DIR",
-	};
+char *get_xdg_pictures_dir(void) {
+	const char *home_dir = getenv("HOME");
+	if (home_dir == NULL) {
+		return NULL;
+	}
+
+	char *config_file;
+	const char *config_home = getenv("XDG_CONFIG_HOME");
+	if (config_home == NULL || config_home[0] == 0) {
+		size_t len = strlen(home_dir) + strlen("/.config/user-dirs.dirs") + 1;
+		config_file = malloc(len);
+		if (config_file == NULL) {
+			return NULL;
+		}
+		strncpy(config_file, home_dir, strlen(home_dir) + 1);
+		strncat(config_file, "/.config/user-dirs.dirs", len);
+	} else {
+		size_t len = strlen(config_home) + strlen("/user-dirs.dirs") + 1;
+		config_file = malloc(len);
+		if (config_file == NULL) {
+			return NULL;
+		}
+		strncpy(config_file, config_home, strlen(config_home) + 1);
+		strncat(config_file, "/user-dirs.dirs", len);
+	}
 
-	for (size_t i = 0; i < sizeof(output_dirs) / sizeof(char *); ++i) {
-		const char *path = getenv(output_dirs[i]);
-		if (path_exists(path)) {
-			return path;
+	FILE *file = fopen(config_file, "r");
+	free(config_file);
+	if (file == NULL) {
+		return NULL;
+	}
+
+	char *line = NULL;
+	size_t line_size = 0;
+	ssize_t nread;
+	char *pictures_dir = NULL;
+	while ((nread = getline(&line, &line_size, file)) != -1) {
+		if (line[nread - 1] == '\n') {
```

Comment:
If the last line is empty, `nread == 0` and this is an out-of-bounds array access. Need to add a `nread > 0` check.

Also I'd prefer to avoid changing the meaning of `nread` in the assignment below: `line[nread - 1]` works too and doesn't mutate `nread`.

Reply:
Done